sklearn/preprocessing/_discretization.py

```diff
@@ -267,7 +278,28 @@ def fit(self, X, y=None):
                 bin_edges[jj] = np.linspace(col_min, col_max, n_bins[jj] + 1)
 
             elif self.strategy == "quantile":
-                quantiles = np.linspace(0, 100, n_bins[jj] + 1)
+                # if sample_weight (weight) parameter is defined, then the size
+                # of each bin should be proportional to each weight value (in w)
+                if sample_weight is not None:
+                    weights = (
```

Comment:
You should use `sklearn.utils.validation._check_sample_weight` to validate sample-weight here.

Reply:
I corrrected this in the following commit : https://github.com/scikit-learn/scikit-learn/pull/22048/commits/85fe2b74ee5fe571b92f113f9dd8182339a4c4aa

I added a check with `_check_sample_weight` on line 267 in _discretization.py